### ISSUE TITLE
创建视图，字段类型显示问题

### ISSUE THREAD

**Issue by awk333** [StarRocks/starrocks#13042]
starrocks 版本 2.2.4

-- 创建视图，多表union的视图
create view rx_bd_test.tmp_test_001 as
select
custom 
, null as user_id
from rx_bd_1000000.dwd_fact_event_0preset_paid_17
union
select
null as custom 
,user_id
from rx_bd_1000000.dwd_fact_event_0preset_share_get_data_15
;

-- 查询视图的列属性
show columns from rx_bd_test.tmp_test_001;
![image](https://user-images.githubusercontent.com/44765875/200238442-ee5f7ac6-3c7f-41bf-842b-ba01d003b254.png)

发现对于user_id的字段类型是null_type，并没有根据 rx_bd_1000000.dwd_fact_event_0preset_share_get_data_15 这个表的user_id来判断类型。

这个问题有可能解决吗？这边需要获取字段类型来做对应的计算操作。

**Comment by Astralidea** [comment]
Can you give us the table creation statement for the base table? We want to know what type the base table corresponds to.

**Comment by Astralidea** [comment]
I use following table and reproduce the problem:
```
CREATE TABLE dwd_fact_event_0preset_paid_17 (
k1 INT,
custom INT,
user_id INT)
DUPLICATE KEY(k1)
PARTITION BY RANGE(k1)
(
PARTITION p1 VALUES LESS THAN ('2')
)
DISTRIBUTED BY HASH(k1)
PROPERTIES (
"replication_num" = "1"
);

CREATE TABLE dwd_fact_event_0preset_share_get_data_15 (
k1 INT,
custom INT,
user_id INT)
DUPLICATE KEY(k1)
PARTITION BY RANGE(k1)
(
PARTITION p1 VALUES LESS THAN ('2')
)
DISTRIBUTED BY HASH(k1)
PROPERTIES (
"replication_num" = "1"
);
```
and this already fix in 2.2.10, You can upgrade to this version and delete the view and rebuild.
